recipes/qt/6.x.x/conanfile.py

```diff
@@ -896,6 +899,8 @@ def _create_plugin(pluginname, libname, type, requires):
                 self.cpp_info.components["qtNetwork"].frameworks.append("GSS")
                 if self.options.gui and self.options.widgets:
                     self.cpp_info.components["qtPrintSupport"].system_libs.append("cups")
+                if self.options.gui and self.options.with_vulkan:
+                    self.cpp_info.components["qtGui"].requires.append("moltenvk::moltenvk")
```

Comment:
Please, move this logic far above where you can see:

```python
            if self.options.get_safe("with_vulkan", False):
                gui_reqs.append("vulkan-loader::vulkan-loader")
```

It will avoid to scatter logics which are tightly coupled.